packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2915,6 +2916,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             throw new UsageError("can't get state when offline load disabled");
         }
 
+        const pendingMessages = this.pendingStateManager.getLocalState();
+        const firstMessage = pendingMessages?.pendingStates[0] as IPendingMessage;
+        const isFirstMessageBatchFirst = firstMessage.opMetadata?.batch;
+        const lastMessage = pendingMessages?.pendingStates[pendingMessages.pendingStates.length - 1].type;
+        assert(!!isFirstMessageBatchFirst, "First message must have batch metadata");
+        assert(isFirstMessageBatchFirst === true, "First message must be first in batch");
+        assert(lastMessage === "flush", "Last message in batch must be flush");
```

Comment:
would be nice to have some tests for this. Call it during a batch with flushmode turnbased, outside a batch, inside ordersequentially with flushmode immediate, etc..